### ISSUE TITLE
关于推理之后的效果的问题

### ISSUE THREAD

**Issue by TK-Operator** [AndSonder/HandWritingEraser-Pytorch#4]
感谢您出彩的工作！
我按照您的环境进行了推理任务，我们的目标是去除试卷上的笔迹，但是为什么生成之后的图片好像没有实现去除笔迹的效果？我倒是看到了手写体与印刷体分离的效果了，以下是一些推理结果：
![Uploading ![5_overlay](https://github.com/user-attachments/assets/658bc191-cc20-4044-86e5-467ca8191b7e)
![5_pred](https://github.com/user-attachments/assets/d02b2407-d64a-4733-ba7d-ecf17ab34220)
![5_target](https://github.com/user-attachments/assets/d6db1535-80b6-415b-a725-7bc122670d08)
![18_image](https://github.com/user-attachments/assets/c09904b3-3a0c-4a6d-8fb0-b116af767b8f)
![18_overlay](https://github.com/user-attachments/assets/0869d57c-d19e-4744-95aa-5bad5a812a4c)
![18_pred](https://github.com/user-attachments/assets/1ee7690a-0c4d-4764-8eba-5da8c05a8084)
![18_target](https://github.com/user-attachments/assets/9c0d7654-4644-4e63-9805-4287a63a8812)
期待您的回复！

**Comment by AndSonder** [comment]
可以后处理一下，转单通道图之后根据像素值做个 mask 就可以分离了

**Comment by TK-Operator** [comment]
感谢您的回复！明白了！然后就是还有两个点不太清楚：
1.我已经训练好了模型，并用其进行了推理任务，但是训练集和测试集都是同一个数据集对吗？而不是分离的，那么在实际测试时，是不是使用的训练时的数据集？这样会不会过拟合？
2.我想使用自己制作的图片进行推理，但是我的图片没有mask、label这两种格式，好像会报错，这意味着如果我要用自己的图片推理，我需要制作这两种格式的图片吗？
感谢您的回复！






------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "AndSonder/HandWritingEraser-Pytorch"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2025年1月15日(星期三) 晚上10:31
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [AndSonder/HandWritingEraser-Pytorch] 关于推理之后的效果的问题 (Issue #4)





 
可以后处理一下，转单通道图之后根据像素值做个 mask 就可以分离了

**Comment by SilentZhang** [comment]
@550developer 
向您请教一下下载完了模型和dataset放到哪里才能运行呢，我完全是一个初学者，大模型训练知识等于0.

**Comment by TK-Operator** [comment]
> [@550developer](https://github.com/550developer) 向您请教一下下载完了模型和dataset放到哪里才能运行呢，我完全是一个初学者，大模型训练知识等于0.

其实我也是新手，不过我知道一点点。不知道您是否有使用服务器？我是将代码下载放在了服务器端运行，然后在服务器进行训练和推理任务，您可以去搜一下如何在服务器或本地使用

**Comment by SilentZhang** [comment]
> > [@550developer](https://github.com/550developer) 向您请教一下下载完了模型和dataset放到哪里才能运行呢，我完全是一个初学者，大模型训练知识等于0.
> 
> 其实我也是新手，不过我知道一点点。不知道您是否有使用服务器？我是将代码下载放在了服务器端运行，然后在服务器进行训练和推理任务，您可以去搜一下如何在服务器或本地使用

我是在本地运行，主要是不知道数据都放在哪个目录下
现在是按照这样的结构把示例文件都放进去了，但是好像不行。
├─data
├─HandWriting
│  ├─Images
│  │  ├─dehw_testA_dataset
│  │  │  └─images
│  │  └─dehw_train_dataset
│  │      ├─gts
│  │      ├─images
│  │      └─mask
│  └─Labels
│      ├─dehw_testA_dataset
│      │  └─images
│      └─dehw_train_dataset
│          ├─gts
│          ├─images
│          └─mask

**Comment by TK-Operator** [comment]
> > > [@550developer](https://github.com/550developer)向您请教一下下载完了模型和数据集放到哪里才能运行呢，我完全是一个初学者，大模型训练知识等于0。
> > 
> > 
> > 其实我也是新手，不过我知道一点。不知道你是否有使用服务器？我只需下载代码放在服务器端运行，然后在服务器上进行训练和推理任务，你可以去搜一下如何在服务器或本地使用
> 
> 我是在本地运行，主要是不知道数据都放在哪个目录下 现在是按照这样的结构把样本文件都渴了，但是没了。 ├─data ├─HandWriting │ ├─Images │ │ ├─dehw_testA_dataset │ │ │ └─images │ │ └─dehw_train_dataset │ │ ├─gts │ │ ├─图像 │ │ └─掩模 │ └─标签 │ ├─dehw_testA_dataset │ │ └─图像 │ └─dehw_train_dataset │ ├─gts │ ├─图像 │ └─掩模
看了你的数据集，应该要下载作者下面给的那个链接里的数据集，不是用百度的那个
